network-libp2p/src/dispatch/codecs/typed.rs

```diff
@@ -8,50 +8,33 @@
 
 use std::{fmt::Debug, io};
 
-use futures::{AsyncRead, AsyncWrite, AsyncWriteExt};
-use libp2p::{
-    core::{upgrade, ProtocolName},
-    request_response::RequestResponseCodec,
-};
-
-use crate::REQRES_PROTOCOL;
+use futures::{AsyncRead, AsyncReadExt, AsyncWrite, AsyncWriteExt};
+use libp2p::{request_response, StreamProtocol};
 
 /// Maximum request size in bytes (5 kB)
-const MAX_REQUEST_SIZE: usize = 5 * 1024;
+const MAX_REQUEST_SIZE: u64 = 5 * 1024;
 /// Maximum response size in bytes (10 MB)
-const MAX_RESPONSE_SIZE: usize = 10 * 1024 * 1024;
+const MAX_RESPONSE_SIZE: u64 = 10 * 1024 * 1024;
 
 #[derive(Clone, Debug, Default)]
 pub struct MessageCodec {}
 
 pub type IncomingRequest = Vec<u8>;
 pub type OutgoingResponse = Vec<u8>;
 
-#[derive(Debug, Clone)]
-pub enum ReqResProtocol {
-    Version1,
-}
-
-impl ProtocolName for ReqResProtocol {
-    fn protocol_name(&self) -> &[u8] {
-        match *self {
-            ReqResProtocol::Version1 => REQRES_PROTOCOL,
-        }
-    }
-}
-
 #[async_trait::async_trait]
-impl RequestResponseCodec for MessageCodec {
-    type Protocol = ReqResProtocol;
+impl request_response::Codec for MessageCodec {
+    type Protocol = StreamProtocol;
     type Request = IncomingRequest;
     type Response = OutgoingResponse;
 
     async fn read_request<T>(&mut self, _: &Self::Protocol, io: &mut T) -> io::Result<Self::Request>
     where
         T: AsyncRead + Unpin + Send,
     {
-        let bytes = upgrade::read_length_prefixed(io, MAX_REQUEST_SIZE).await?;
```

Comment:
AFAICT, this is changing the network in an incompatible way, is this desired?

Reply:
Yeah, I didn't think it through. I changed it back to be compatible. Unfortunately libp2p no longer support these functions so needed to me it to our code